assets/wizards/connections/views/main/mailchimp.js

```diff
@@ -0,0 +1,172 @@
+		setIsLoading( true );
+		apiFetch( {
+			path: '/newspack/v1/oauth/mailchimp',
+			method: 'POST',
+			data: {
+				api_key: apiKey,
+			},
+		} )
+			.then( res => {
+				setAuthState( res );
+			} )
+			.catch( e => {
+				setError(
+					e.message ||
+						__( 'Something went wrong during verification of your Mailchimp API key.', 'newspack' )
+				);
+			} )
+			.finally( () => {
+				setIsLoading( false );
+				closeModal();
+			} );
+	};
+
+	const disconnect = () => {
+		setIsLoading( true );
+		apiFetch( {
+			path: '/newspack/v1/oauth/mailchimp',
+			method: 'DELETE',
+		} )
+			.then( () => {
+				setAuthState( {} );
+				setIsLoading( false );
+			} )
+			.catch( handleError );
+	};
+
+	const getDescription = () => {
+		if ( isLoading ) {
+			return __( 'Loading…', 'newspack' );
+		}
+		if ( isConnected ) {
+			return sprintf( __( 'Connected as %s', 'newspack' ), authState.username );
+		}
+		return __( 'Not connected', 'newspack' );
+	};
+
+	const getModalButtonText = () => {
+		if ( isLoading ) {
+			return __( 'Connecting…', 'newspack' );
+		}
+		if ( isConnected ) {
+			return __( 'Connected', 'newspack' );
+		}
+		return __( 'Connect', 'newspack' );
+	};
+
+	return (
+		<>
+			<ActionCard
+				title="Mailchimp"
+				description={ getDescription() }
+				checkbox={ isConnected ? 'checked' : 'unchecked' }
+				actionText={
+					<Button
+						isLink
+						isDestructive={ isConnected }
+						onClick={ isConnected ? disconnect : openModal }
+						disabled={ isLoading }
+					>
+						{ isConnected ? __( 'Disconnect', 'newspack' ) : __( 'Connect', 'newspack' ) }
+					</Button>
+				}
+				isMedium
+			/>
+			{ isModalOpen && (
+				<Modal title={ __( 'Add Mailchimp API Key', 'newspack' ) } onRequestClose={ closeModal }>
+					<div ref={ modalTextRef }>
+						<Grid columns={ 1 } gutter="0">
+							<TextControl
+								placeholder="1234a56b7cde89f0g1h234-us1"
+								label={ __( 'Mailchimp API Key', 'newspack' ) }
+								hideLabelFromVision={ true }
+								value={ apiKey }
+								onChange={ setAPIKey }
+								onKeyDown={ event => {
+									if ( ENTER === event.keyCode && '' !== apiKey ) {
+										event.preventDefault();
+										submitAPIKey();
+									}
+								} }
+							/>
+							<Card noBorder>
+								<ExternalLink href="https://us1.admin.mailchimp.com/account/api/">
+									{ __( 'Generate Mailchimp API key', 'newspack' ) }
+								</ExternalLink>
+								<span className="sep sep__inline" />
+								<ExternalLink href="https://mailchimp.com/help/about-api-keys/">
+									{ __( 'About Mailchimp API keys', 'newspack' ) }
+								</ExternalLink>
+							</Card>
+						</Grid>
+					</div>
+					<Card buttonsCard noBorder className="justify-end">
+						<Button
+							isSecondary
+							onClick={ () => {
+								closeModal();
+							} }
+						>
+							{ __( 'Cancel', 'newspack' ) }
+						</Button>
+						<Button isPrimary disabled={ ! apiKey } onClick={ () => submitAPIKey() }>
```

Comment:
Nit ;) And same with a few other cases in this file.

```suggestion
						<Button isPrimary disabled={ ! apiKey } onClick={ submitAPIKey }>
```